WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/CoverBlockProcessor.java

```diff
@@ -0,0 +1,78 @@
+package org.wordpress.android.ui.posts.mediauploadcompletionprocessors;
+
+import com.google.gson.JsonObject;
+
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.wordpress.android.util.helpers.MediaFile;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class CoverBlockProcessor extends BlockProcessor {
+    /**
+     * Template pattern used to match and splice cover inner blocks
+     */
+    private static final Pattern PATTERN_COVER_INNER = Pattern.compile(new StringBuilder()
+                    .append("(^.*?<div class=\"wp-block-cover__inner-container\">\\s*)")
+                    .append("(.*)") // inner block contents
+                    .append("(\\s*</div>\\s*</div>\\s*<!-- /wp:cover -->.*)").toString(), Pattern.DOTALL);
+
+    /**
+     * Pattern to match background-image url in cover block html content
+     */
+    private static final Pattern PATTERN_BACKGROUND_IMAGE_URL = Pattern.compile("background-image:url\\([^\\)]+\\)");
```

Comment:
so definitely there's an issue with blocks defining their attributes in the "JSON attributes" vs. inside HTML, but instead of hard-coding a string-matching pattern we can go back to `jsoup` as you were doing and hard-code the [known attribute definition](https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/cover/block.json), which would then end up looking like this…

```java
JSONAttribute backgroundImageUrl = new JSONAttribute("url", JSONAttribute.String);
```

I'm sure there's a million ways to do this. The main point I was trying to make is that we can rely on the structural form of the data instead of processing the HTML string as if it were plain text. It's quite possible to have `background-image:url` inside the block in a way that isn't meant to indicate a different background, and this string-matching might end up paving a road of headaches to debug 😄

Reply:
> It's quite possible to have `background-image:url` inside the block in a way that isn't meant to indicate a different background, and this string-matching might end up paving a road of headaches to debug

Ah, indeed! I wonder if there is a misunderstanding about how this pattern is currently used. This only matches on the `style` attribute [within the parsed HTML of a matching block](https://github.com/wordpress-mobile/WordPress-Android/pull/11541/files/e6c78af481ba7d6d361d9ff526933494bac0ee32#diff-8c06e672b915270d7915261d8e7e367cR68), and so the string replacement is restricted to that particular inline style. In other words, it will _not_ replace similar text within the content of HTML (e.g. a caption, a paragraph, etc.). Perhaps the Javadoc could be made more specific.